### ISSUE TITLE
`isEqual` in analytics-connector has a bug comparing `null`

### ISSUE THREAD

**Issue by sod** [amplitude/experiment-js-client#41]
throw the isEqual method into the console https://github.com/amplitude/experiment-js-client/blob/main/packages/analyticsConnector/src/identityStore.ts#L130-L176 and run

```js
isEqual({userId: null}, {userId: null})
```

and you see the error
```log
VM229:14 Uncaught TypeError: Cannot read properties of null (reading 'length')
```

<img width="652" alt="CleanShot 2022-09-06 at 12 52 00@2x" src="https://user-images.githubusercontent.com/905328/188617168-ee8f1ee6-ebfc-4869-ba86-d89d2d7e79e9.png">

This is causing amplitude-js to throw with the same error if you do `getInstance().setUserId(null)`

**Comment by bgiori** [comment]
fixed in `v1.5.6`. Thanks for reporting this @sod !

**Comment by sod** [comment]
Thanks a lot for the quick fix ❤️ 

Will the https://www.npmjs.com/package/@amplitude/analytics-connector in version 1.4.6 - the package where the fix happened - published to npm?
